sync/class.jetpack-sync-full.php

```diff
@@ -337,22 +336,35 @@ public function expand_updates( $args ) {
 		return $args;
 	}
 	
-	function update_sent_progress_action( $action_to_send ) {
+	function update_sent_progress_action( $actions_sent ) {
 		$modules_count = array();
 		$status = $this->get_status();
 		if ( is_null( $status['started'] ) || $status['finished'] ) {
 			return;
 		}
-		foreach( $action_to_send as $action ) {
+
+		if ( in_array( 'jetpack_full_sync_start', $actions_sent ) ) {
```

Comment:
Oh! I see. It seems like the hook `jetpack_sync_processed_actions` would be more useful if it included the arguments as well as the action name - it also would work more like other, similarly-named actions. You'd then be able to avoid creating that extra array as well.